src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -193,6 +207,23 @@ describe('insights', () => {
       expect(getUserToken()).toEqual('abc');
     });
 
+    it('applies userToken without resetting the page', () => {
+      const {
+        insightsClient,
+        instantSearchInstance,
+        getUserToken,
+        helper,
+      } = createTestEnvironment();
+      const middleware = createInsightsMiddleware({
+        insightsClient,
+      })({ instantSearchInstance });
+      insightsClient('setUserToken', 'abc');
```

Comment:
Not a big difference but what if this line is below `middleware.subscribe()`?

```js
      helper.setPage(100);
      middleware.subscribe();
      expect(helper.state.page).toBe(100); 
```

is already proved correct in the previous test, so it'd make sense to set user token after those, and see if it still doesn't change the page.